src/Common/HTTPConnectionPool.cpp

```diff
@@ -393,6 +450,9 @@ class EndpointConnectionPool : public std::enable_shared_from_this<EndpointConne
                 }
             }
             response_stream = nullptr;
+            // FIXME: We are not sure that response stream is fully read at this moment, so hooks could possible be called after this point, right?
+            // Session::setSendDataHooks();
+            // Session::setReceiveDataHooks();
```

Comment:
```suggestion
```

Reply:
No, they could not. Connection is destroyed. I do not find code where they are preserved for the next reuse, which is right.